datadog_checks_downloader/tests/test_downloader.py

```diff
@@ -37,39 +64,289 @@
 ]
 
 
+@pytest.fixture(autouse=True)
+def _clear_cache():
+    """Clear the downloader's cache to make sure it does not affect test results."""
+    # current
+    metadata_current_dir = os.path.join(
+        os.path.dirname(datadog_checks.downloader.__file__),
+        "data",
+        "repo",
+        "metadata",
+        "current",
+    )
+    for file_ in os.listdir(metadata_current_dir):
+        if file_ == "root.json" or file_.startswith("."):
+            continue
+
+        file_path = os.path.join(metadata_current_dir, file_)
+
+        if not os.path.isfile(file_path):
+            # Skip any nested dirs.
+            continue
+
+        os.remove(file_path)
+
+    # previous
+    metadata_previous_dir = os.path.join(
+        os.path.dirname(datadog_checks.downloader.__file__),
+        "data",
+        "repo",
+        "metadata",
+        "previous",
+    )
+    for file_ in os.listdir(metadata_previous_dir):
+        if file_.startswith("."):
+            continue
+
+        file_path = os.path.join(metadata_previous_dir, file_)
+
+        if not os.path.isfile(file_path):
+            # Skip any nested dirs.
+            continue
+
+        os.remove(file_path)
+
+    # targets
+    targets_dir = os.path.join(os.path.dirname(datadog_checks.downloader.__file__), "data", "repo", "targets")
+    for item in os.listdir(targets_dir):
+        if item.startswith("."):
+            continue
+
+        shutil.rmtree(os.path.join(targets_dir, item))
+
+
+def _do_run_downloader(argv):
+    """Run the Datadog checks downloader."""
+    old_sys_argv = sys.argv
+
+    sys.argv = ["datadog_checks_downloader"] + argv  # Make sure argv[0] (program name) is prepended.
+    try:
+        download()
+    finally:
+        sys.args = old_sys_argv
+
+
+@pytest.mark.online
+def test_download(capfd, distribution_name, distribution_version):
+    """Test datadog-checks-downloader successfully downloads and validates a wheel file."""
+    argv = [distribution_name, "--version", distribution_version]
+
+    _do_run_downloader(argv)
+    stdout, stderr = capfd.readouterr()
+
+    assert not stderr, "No standard error expected, got: {}".format(stderr)
+
+    output = [line for line in stdout.splitlines() if line]
+    assert len(output) == 1, "Only one output line expected, got {}:\n\t{}".format(len(output), stdout)
+
+    # XXX: could be extended to be less error-prone.
+    delimiter = "datadog_checks_downloader/datadog_checks/downloader/data/repo/targets/" "simple/{}/{}-{}".format(
+        distribution_name, distribution_name.replace("-", "_"), distribution_version
+    )
```

Comment:
Just curious about this, but is `delimiter` a tuple of two strings? I'm not sure how you can have two strings but they aren't concatenated?

Reply:
This one was probably brought by formatting or IDE. Strings are concatenated here, there is no tuple. Will fix this, you have thorough reviews 👍🏻 

```python
>>> " one " " two " " three "
' one  two  three '
```

Reply:
Adjusted in 72022518b3